src/Microsoft.TemplateEngine.Cli/HelpAndUsage/HelpForTemplateResolution.cs

```diff
@@ -8,82 +8,69 @@
-                {
-                    // this is really an error
-                    unambiguousTemplateGroupForList = new List<ITemplateMatchInfo>();
-                }
-
-                if (templateResolutionResult.UsingPartialMatches)
-                {
-                    ShowNoTemplatesFoundMessage(commandInput.TemplateName, commandInput.Language, commandInput.TypeFilter);
-                    return CreationResultStatus.NotFound;
-                }
-
-                ShowTemplatesFoundMessage(commandInput.TemplateName, commandInput.Language, commandInput.TypeFilter);
-                DisplayTemplateList(unambiguousTemplateGroupForList, environmentSettings, commandInput.Language, defaultLanguage);
-                // list flag specified, so no usage examples or detailed help
-                return CreationResultStatus.Success;
+                IReadOnlyCollection<ITemplateMatchInfo> unambiguousTemplateGroupForDetailDisplay = templateResolutionResult.UnambiguousTemplateGroup;
+                return TemplateDetailedHelpForSingularTemplateGroup(unambiguousTemplateGroupForDetailDisplay, environmentSettings, commandInput, hostDataLoader, templateCreator);
             }
-            else
+            //if language is not specified and group has template that matches the language show the help for that the template that matches the language
+            if (string.IsNullOrEmpty(commandInput.Language) && !string.IsNullOrEmpty(defaultLanguage) && templateResolutionResult.HasUnambiguousTemplateGroupForDefaultLanguage)
             {
-                // not in list context, but Unambiguous
-                // this covers whether or not --help was input, they do the same thing in the unambiguous case
+                IReadOnlyCollection<ITemplateMatchInfo> unambiguousTemplateGroupForDetailDisplay = templateResolutionResult.UnambiguousTemplatesForDefaultLanguage;
                 return TemplateDetailedHelpForSingularTemplateGroup(unambiguousTemplateGroupForDetailDisplay, environmentSettings, commandInput, hostDataLoader, templateCreator);
             }
+            else
+            {
+                return DisplayListOrHelpForAmbiguousTemplateGroup(templateResolutionResult, environmentSettings, commandInput, hostDataLoader, telemetryLogger, defaultLanguage);
+            }
         }
+       
 
-        private static CreationResultStatus TemplateDetailedHelpForSingularTemplateGroup(IReadOnlyList<ITemplateMatchInfo> unambiguousTemplateGroup, IEngineEnvironmentSettings environmentSettings, INewCommandInput commandInput, IHostSpecificDataLoader hostDataLoader, TemplateCreator templateCreator)
+        private static CreationResultStatus TemplateDetailedHelpForSingularTemplateGroup(IReadOnlyCollection<ITemplateMatchInfo> unambiguousTemplateGroup, IEngineEnvironmentSettings environmentSettings, INewCommandInput commandInput, IHostSpecificDataLoader hostDataLoader, TemplateCreator templateCreator)
```

Comment:
minor: the scheme does not seem very Csharpie.  Would an enum listing the distinct singular template group items simplify the logic distributed across these methods?

Reply:
Not sure what I understood the change suggested here. unambiguousTemplateGroup enum already lists template group to show. In most cases this enum contains only one element, but might also contain several then the highest precedence template is shown. 
I have the question about template group and their usages for Kathleen and want to understand if the case of having several templates in the group at this point of time is possible and should be supported.